users/api/endpoint.go

```diff
@@ -31,6 +31,25 @@ func registrationEndpoint(svc users.Service) endpoint.Endpoint {
 	}
 }
 
+func createUserEndpoint(svc users.Service) endpoint.Endpoint {
```

Comment:
What's this? Don't we already have an endpoint to create users?

Reply:
Yes, we have an endpoint to create users. However, we thought that only admin users should be able to create users. So, instead of `Register`, I created a `CreateUser` endpoint which verifies if the user who called `CreateUser` has rights to create a new user, and writes a default policy for the newly created user. The old endpoint still remains for the admin creation as here https://github.com/mainflux/mainflux/blob/19f0437f577fa24ae0fe280fda3a0cf0a6a1c674/cmd/users/main.go#L323.

Reply:
Ah I see. I would demand one more significant change then - now that we are introducing policies and admins, I think we should by default support only creating users via admin, and not self-sign-on. Self-sign-on should be an option, that can be activated via ENV var, and should have a dedicated endpoint - something like `/selfsignon` or something. And we should keep simple `POST /users` user creation endpoint to be used by the admin - i.e. we should add authorization to it (currently it is not authorized, because it is self-sign-on).